pkg/cmd/server/origin/ensure.go

```diff
@@ -22,3 +22,11 @@ func (c *MasterConfig) ensureOpenShiftSharedResourcesNamespace(context genericap
 	}
 	return nil
 }
+
+// TODO: We should have a single place where we register all post start hook
+// and make this function unnecessary
+func ensureAddPostStartHook(genericAPIServer *genericapiserver.GenericAPIServer, name string, hook genericapiserver.PostStartHookFunc) {
```

Comment:
Move this upstream.  `AddPostStartHookOrDie`.  We want the separate function with an error to allow testing of the error condition (collision on names as I recall).